tests/neo4j/test_bookmarks.py

```diff
@@ -72,3 +72,136 @@ def test_no_bookmark_after_rollback(self):
+                         exc.exception.code)
+
+    def test_can_handle_multiple_bookmarks(self):
+        bookmarks = []
+        expected_node_count = 5
+        test_execution_id = uuid4().hex
+
+        def create_node(tx):
+            result = tx.run(
+                "CREATE (t:MultipleBookmarksTest {testId:$uuid})",
+                params={"uuid": types.CypherString(test_execution_id)})
+            result.consume()
+
+        for _i in range(0, expected_node_count):
+            self._session = self._driver.session("w")
+            self._session.write_transaction(create_node)
+            bookmarks.append(self._session.last_bookmarks())
+            self._session.close()
+
+        bookmarks = [bookmark for sublist in bookmarks for bookmark in sublist]
+        self._session = self._driver.session("r", bookmarks)
+
+        def get_node_count(tx):
+            result = tx.run(
+                "MATCH (t:MultipleBookmarksTest {testId:$uuid})"
+                " RETURN count(t)",
+                params={"uuid": types.CypherString(test_execution_id)})
+            record = result.next()
+            return record.values[0]
+
+        count = self._session.read_transaction(get_node_count)
+        self.assertEqual(types.CypherInt(expected_node_count), count)
+
+    @cluster_unsafe_test
+    def test_can_pass_write_bookmark_into_write_session(self):
+        test_execution_id = uuid4().hex
+        self._session = self._driver.session("w")
+        tx = self._session.begin_transaction()
+        result = tx.run(
+            "CREATE (t:AccessModeTest {testId:$uuid})",
+            params={"uuid": types.CypherString(test_execution_id)})
+        result.consume()
+        tx.commit()
+        bookmarks = self._session.last_bookmarks()
+        self._session.close()
+
+        self._session = self._driver.session("w", bookmarks)
+        tx = self._session.begin_transaction()
+        result = tx.run(
+            "MATCH (t:AccessModeTest {testId:$uuid})"
+            " RETURN count(t)",
+            params={"uuid": types.CypherString(test_execution_id)})
+        record = result.next()
+        node_count = record.values[0]
+        tx.commit()
+
+        self.assertEqual(types.CypherInt(1), node_count)
+
+    @cluster_unsafe_test
+    def test_can_pass_read_bookmark_into_write_session(self):
+        test_execution_id = uuid4().hex
+        self._session = self._driver.session("w")
+        tx = self._session.begin_transaction()
+        result = tx.run(
+            "CREATE (t:AccessModeTest {testId:$uuid})",
+            params={"uuid": types.CypherString(test_execution_id)})
+        result.consume()
+        tx.commit()
+        bookmarks = self._session.last_bookmarks()
+        self._session.close()
+
+        self._session = self._driver.session("r", bookmarks)
+        tx = self._session.begin_transaction()
+        result = tx.run(
+            "MATCH (t:AccessModeTest {testId:$uuid})"
+            " RETURN count(t)",
```

Comment:
```suggestion
            "MATCH (t:AccessModeTest {testId:$uuid}) "
            "RETURN count(t)",
```
no deal breaker but for more consistency.